tket/tests/test_Synthesis.cpp

```diff
@@ -2186,5 +2186,124 @@ SCENARIO("Restricting ZZPhase gate angles.") {
 
   REQUIRE(comparison == c);
 }
+
+SCENARIO("Test decompose_ZXZ_to_TK1") {
+  Circuit circ;
+  GIVEN("A simple ZZ circuit") {
+    circ = Circuit(1);
+    circ.add_op<unsigned>(OpType::Rz, 0.234, {0});
+    circ.add_op<unsigned>(OpType::Rz, 0.434, {0});
+  }
+  GIVEN("A simple XX circuit") {
+    circ = Circuit(1);
+    circ.add_op<unsigned>(OpType::Rx, 0.234, {0});
+    circ.add_op<unsigned>(OpType::Rx, 0.434, {0});
+  }
+  GIVEN("A simple ZXZ circuit") {
+    circ = Circuit(1);
+    circ.add_op<unsigned>(OpType::Rz, 0.234, {0});
+    circ.add_op<unsigned>(OpType::Rx, 1.334, {0});
+    circ.add_op<unsigned>(OpType::Rz, 0.434, {0});
+  }
+  GIVEN("A simple ZXZ circuit with global phases") {
+    circ = Circuit(1);
+    circ.add_op<unsigned>(OpType::Rz, 2.234, {0});
+    circ.add_op<unsigned>(OpType::Rx, 3.334, {0});
+    circ.add_op<unsigned>(OpType::Rz, 2.434, {0});
+  }
+  GIVEN("A circuit with irreducible gates") {
+    circ = Circuit(2);
+    circ.add_op<unsigned>(OpType::Rz, 2.234, {0});
+    circ.add_op<unsigned>(OpType::Rx, 3.334, {0});
+    circ.add_op<unsigned>(OpType::V, {0});
+    circ.add_op<unsigned>(OpType::Rz, 2.434, {0});
+    circ.add_op<unsigned>(OpType::Rz, 12.23, {1});
+    circ.add_op<unsigned>(OpType::Sdg, {1});
+    circ.add_op<unsigned>(OpType::Rx, 22.22, {1});
+  }
+  GIVEN("A circuit with irreducible gates and blocking multiqb gates") {
+    circ = Circuit(2);
+    circ.add_op<unsigned>(OpType::Rz, 2.234, {0});
+    circ.add_op<unsigned>(OpType::CX, {0, 1});
+    circ.add_op<unsigned>(OpType::Rx, 3.334, {0});
+    circ.add_op<unsigned>(OpType::Rz, 0.123, {0});
+    circ.add_op<unsigned>(OpType::V, {0});
+    circ.add_op<unsigned>(OpType::Rz, 2.434, {0});
+    circ.add_op<unsigned>(OpType::Rz, 12.23, {1});
+    circ.add_op<unsigned>(OpType::T, {0});
+    circ.add_op<unsigned>(OpType::T, {1});
+    circ.add_op<unsigned>(OpType::Sdg, {1});
+    circ.add_op<unsigned>(OpType::Rz, 2.434, {0});
+    circ.add_op<unsigned>(OpType::CZ, {0, 1});
+    circ.add_op<unsigned>(OpType::Rx, 22.22, {1});
+    circ.add_op<unsigned>(OpType::Rx, 3.334, {0});
+  }
+
+  auto u0 = tket_sim::get_unitary(circ);
+  Transforms::decompose_ZXZ_to_TK1().apply(circ);
```

Comment:
Maybe require this to return true and also test that the rebased circuits only have tk1 and 2-q gates? Similar for the ZYZ test below.

Reply:
Done.